### PR TITLE
feat: add ADR Process, inspired by the backstage ADR structure

### PR DESCRIPTION
**Pull request by kallekrantz**

### CODE REVIEW

**Comment by kallekrantz** [comment]
Marking as draft until #206 is merged

**Comment by DavyJ0nes** [comment]
As this is a public repo, does it make sense to add this? I'm not against it but would mean some duplication and sanitisation of internal ADRs related to this repo.

**Comment by kallekrantz** [comment]
@DavyJ0nes Correct, for we're specifically intending this to be ADRs related to the structure of the API, such as moving to V2 and how the choices made have an effect on the API structure.  The idea came from @radhus and I personally really like it, to show a little bit more of a log of API Changes.

File: docs/architecture-decisions/index.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+______________________________________________________________________
+
+| id            | title                               | sidebar_label | description                                     |
+| ------------- | ----------------------------------- | ------------- | ----------------------------------------------- |
+| adrs-overview | Architecture Decision Records (ADR) | Overview      | Overview of Architecture Decision Records (ADR) |
+
+______________________________________________________________________
+
+The architecture decisions made across the Backup feature in Einride are kept here. For more information about ADRs, when to write them, and why, please see [this blog post](https://engineering.atspotify.com/2020/04/14/when-should-i-write-an-architecture-decision-record/).
```
Comment (DavyJ0nes) [inline docs/architecture-decisions/index.md:9]:
```suggestion
The architecture decisions made for the Einride Extend public API are kept here. For more information about ADRs, when to write them, and why, please see [this blog post](https://engineering.atspotify.com/2020/04/14/when-should-i-write-an-architecture-decision-record/).
```

Comment (kallekrantz) [reply docs/architecture-decisions/index.md]: @DavyJ0nes :facepalm: fixed!

**Review by DavyJ0nes — APPROVED** [review]

**Pull request merged**